draft-ietf-httpbis-header-structure.md

```diff
@@ -179,76 +179,67 @@ This specification defines minimums for the length or number of various structur
 This section defines the abstract value types that can be composed into Structured Headers. The ABNF provided represents the on-wire format in HTTP/1.
 
 
-## Dictionaries {#dictionary}
-
-Dictionaries are ordered maps of key-value pairs, where the keys are short, textual strings and the values are items ({{item}}). There can be one or more members, and keys are required to be unique.
+## Lists {#list}
 
-Implementations MUST provide access to dictionaries both by index and by key. Specifications MAY use either means of accessing the members.
+Lists are arrays of items ({{item}}) with one or more members. Members can also be an "inner list", itself an array of items.
 
-The ABNF for dictionaries in HTTP/1 headers is:
+The ABNF for lists in HTTP/1 headers is:
 
 ~~~ abnf
-sh-dictionary  = dict-member *( OWS "," OWS dict-member )
-dict-member    = member-name "=" member-value
-member-name    = key
-member-value   = sh-item
-key            = lcalpha *( lcalpha / DIGIT / "_" / "-" )
-lcalpha        = %x61-7A ; a-z
+sh-list     = list-member *( OWS "," OWS list-member )
+list-member = sh-item / inner-list
```

Comment:
Would it be possible to add param-item as an option here?

Considering the fact that list and parameterized list are the two compound types that can use a `;` separator, I think it would be beneficial to allow both, for example when defining a new Cache-Control directive that takes a parameterized list as an argument (pretending that Cache-Control is a SH header).

Reply:
If we added param-item here, it would have the effect of collapsing parameterised lists into lists -- which is possible, I think (if we could define a sane parsing algorithm that could tell the difference). 

However, that doesn't seem like what you're asking for. Cache-Control maps to a dictionary; if you allowed a dictionary member to be a parameterised item, what would that look like?

Reply:
Doesn't that undo #443 and bring back the [atomic-or-complex problem](https://github.com/httpwg/http-extensions/issues/443#issuecomment-362997206)?

Reply:
@phluid61 Thanks for pointing that out. OTOH, am I correct in assuming that the reason we handled #443 that way is because we did not know of a good way to resolve the issue?

If that is the case, this might be the correct moment to revisit the issue, now that we are leaning towards having surrounding parens.

Reply:
@mnot 
> However, that doesn't seem like what you're asking for. Cache-Control maps to a dictionary;

Apologies. I should have left my comment where the change for the Dictionary is made.

>  if you allowed a dictionary member to be a parameterised item, what would that look like?

What I have in mind is something like `Cache-Control: no-cache, mydirective arg1=a;arg2=b`.

Reply:
I think what you're asking for there essentially is a Parameterised Dictionary. I'd do it as something like 

~~~
Example: foo=1, bar=2;a=b;c=d
~~~

... but that's just a fairly minor syntactic difference. Could you open an issue for that?